packages/core/tests/middleware/state-with-snapshot.spec.ts

```diff
@@ -28,18 +18,16 @@ type TestState = {
 
 describe("state with snapshot middleware", () => {
   test("should preserve state when resuming from snapshot", async () => {
-    // Create the combined middleware like in the demo
-    const stateful = createStatefulMiddleware((state: TestState) => state);
-    const statefulWorkflow = stateful.withState(createWorkflow());
-    const workflow = withSnapshot(statefulWorkflow as unknown as Workflow);
+    const { withState, getContext } = createStatefulMiddleware(
```

Comment:
don't use getContext from the middleware (i updated the getContext from context.ts to contain the right context object. if the types are not working use any as a temp solution)
```suggestion
    const { withState } = createStatefulMiddleware(
```

Reply:
you can actually remove getContext from createStatefulMiddleware

Reply:
sorry, we can remove it in a second PR and keep it as is for now (too many changes)